modules/apps/captcha/captcha-taglib/src/main/java/com/liferay/captcha/taglib/servlet/taglib/CaptchaTag.java

```diff
@@ -50,22 +56,44 @@ protected String getPage() {
 
 	@Override
 	protected void setAttributes(HttpServletRequest httpServletRequest) {
+		httpServletRequest.setAttribute(
+			"liferay-captcha:captcha:portletId",
+			_getPortletId(httpServletRequest));
 		httpServletRequest.setAttribute(
 			"liferay-captcha:captcha:url", _getURL(httpServletRequest));
 	}
 
+	private String _getPortletId(HttpServletRequest httpServletRequest) {
+		return PortalUtil.getPortletId(httpServletRequest);
+	}
+
 	private String _getURL(HttpServletRequest httpServletRequest) {
 		if (Validator.isNotNull(_url)) {
 			return _url;
 		}
 
+		try {
+			CaptchaConfiguration captchaConfiguration =
+				ConfigurationProviderUtil.getSystemConfiguration(
+					CaptchaConfiguration.class);
+
+			if (captchaConfiguration.enableSimpleCaptchaHeadlessAPI()) {
+				return StringPool.BLANK;
```

Comment:
I think the API URL should be returned here instead. What do you think?

Reply:
Fixed it. Thanks